flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala

```diff
@@ -439,17 +439,16 @@ object FlinkRexUtil {
   })
 
   private class EquivalentExprShuttle(rexBuilder: RexBuilder) extends RexShuttle {
-    private val equiExprMap = mutable.HashMap[String, RexNode]()
+    private val equiExprSet = mutable.HashSet[RexNode]()
 
     override def visitCall(call: RexCall): RexNode = {
       call.getOperator match {
         case EQUALS | NOT_EQUALS | GREATER_THAN | LESS_THAN | GREATER_THAN_OR_EQUAL |
             LESS_THAN_OR_EQUAL =>
-          val swapped = swapOperands(call)
-          if (equiExprMap.contains(swapped.toString)) {
-            swapped
+          if (equiExprSet.contains(call)) {
```

Comment:
For Calcite `>($1, $2)` and `<($2, $1)` are equal, same for others
this fact could be used here.

Ideally we could reuse Calcite's RexNormalize however it will lead to large plan related changes, may be makes sense to do it with separate PR/jira

Reply:
Did you already try the RexNormalize change?

That does sound like a good idea / nice change.  Have you filled a JIRA for it?

Reply:
jira issue https://issues.apache.org/jira/browse/FLINK-36367

I tried
1. it touches more than 100 plans which is still doable
2. I faced one float issue for a couple of plans: depending on how tests are running (just one or for the whole class) the order of operands is different (the operator expression is valid for both).
So need more time to debug and fix it

Reply:
> 2. I faced one float issue for a couple of plans: depending on how tests are running (just one or for the whole class) the order of operands is different (the operator expression is valid for both).
>    So need more time to debug and fix it
Interesting.  Is it a bug in RexNormalize or something else?